domains/authorization/Authorization.API/API/Authorization/[Features]/AcceptTermsCommand.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using API.Models;
+using API.Repository;
+using API.ValueObjects;
+using API.Data;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using MassTransit;
+using EnergyOrigin.IntegrationEvents.Events.Terms.V2;
+
+namespace API.Authorization._Features_;
+
+public record AcceptTermsCommand(string OrgCvr, string OrgName, Guid UserId) : IRequest<bool>;
+
+public class AcceptTermsCommandHandler(
+    IOrganizationRepository organizationRepository,
+    ITermsRepository termsRepository,
+    IUnitOfWork unitOfWork,
+    IPublishEndpoint publishEndpoint)
+    : IRequestHandler<AcceptTermsCommand, bool>
+{
+    public async Task<bool> Handle(AcceptTermsCommand request, CancellationToken cancellationToken)
+    {
+        await unitOfWork.BeginTransactionAsync();
+
+        try
+        {
+            var usersOrganizationsCvr = Tin.Create(request.OrgCvr);
+
+            var usersAffiliatedOrganization = await organizationRepository.Query()
+                .FirstOrDefaultAsync(o => o.Tin == usersOrganizationsCvr, cancellationToken);
+
+            if (usersAffiliatedOrganization == null)
+            {
+                usersAffiliatedOrganization = Organization.Create(usersOrganizationsCvr, OrganizationName.Create(request.OrgName));
+                await organizationRepository.AddAsync(usersAffiliatedOrganization, cancellationToken);
+            }
+
+            var latestTerms = await termsRepository.Query()
+                .OrderByDescending(t => t.Version)
```

Comment:
Since Version is a string column, this must mean "3.0" > "10.0" ? Is this expected?

Reply:
fixed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/e9c31673e661c9e2063fac0775e3e70940d7e0eb